### ISSUE TITLE
Windows release failed rosidl_generator_py 'test_interfaces_py'

### ISSUE THREAD

**Issue by clalancette** [ros2/build_farmer#209]
Last night, the Windows release build failed test_interfaces_py: https://ci.ros2.org/view/nightly/job/nightly_win_rel/1201/testReport/junit/projectroot/rosidl_generator_py/test_interfaces_py/

The error message is:
```
 - C:/Python37/python.exe -u -m pytest C:/J/workspace/nightly_win_rel/ws/src/ros2/rosidl_python/rosidl_generator_py/test/test_interfaces.py -o cache_dir=C:/J/workspace/nightly_win_rel/ws/build/rosidl_generator_py/ament_cmake_pytest/test_interfaces_py/.cache --junit-xml=C:/J/workspace/nightly_win_rel/ws/build/rosidl_generator_py/test_results/rosidl_generator_py/test_interfaces_py.xunit.xml --junit-prefix=rosidl_generator_py
Traceback (most recent call last):
  File "C:\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python37\lib\site-packages\pytest.py", line 7, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "C:\Python37\lib\site-packages\_pytest\assertion\__init__.py", line 12, in <module>
    from _pytest.assertion import rewrite
  File "C:\Python37\lib\site-packages\_pytest\assertion\rewrite.py", line 23, in <module>
    from _pytest.assertion import util
  File "C:\Python37\lib\site-packages\_pytest\assertion\util.py", line 10, in <module>
    import _pytest._code
  File "C:\Python37\lib\site-packages\_pytest\_code\__init__.py", line 6, in <module>
    from .code import Code  # noqa
  File "C:\Python37\lib\site-packages\_pytest\_code\code.py", line 14, in <module>
    import pluggy
  File "C:\Python37\lib\site-packages\pluggy\__init__.py", line 16, in <module>
    from .manager import PluginManager, PluginValidationError
  File "C:\Python37\lib\site-packages\pluggy\manager.py", line 6, in <module>
    import importlib_metadata
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 463, in <module>
    __version__ = version(__name__)
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 430, in version
    return distribution(package).version
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 403, in distribution
    return Distribution.from_name(package)
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 174, in from_name
    dist = next(dists, None)
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 360, in <genexpr>
    for path in map(cls._switch_path, paths)
  File "C:\Python37\lib\site-packages\importlib_metadata\__init__.py", line 375, in _search_path
    if not root.is_dir():
  File "C:\Python37\lib\pathlib.py", line 1351, in is_dir
    return S_ISDIR(self.stat().st_mode)
  File "C:\Python37\lib\pathlib.py", line 1161, in stat
    return self._accessor.stat(self)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\J\\workspace\\nightly_win_rel\\ws\\build\\rosidl_generator_py\\rosidl_generator_py\\"C:\\J\\workspace\\nightly_win_rel\\ws\\build\\rosidl_generator_py\\rosidl_generator_py:C:\\J\\workspace\\nightly_win_rel\\ws\\src\\ros2\\rosidl_python\\rosidl_generator_py"'
```

**Comment by dirk-thomas** [comment]
The chain of thoughts so far:
* `pytest` uses `pluggy`
* `pluggy` released a new version yesterday: 0.12.0
* that version starts to use `importlib_metadata`
in which the exception happens - as far as I can see by feeding an invalid path in the variable `root`.

To be continued...

**Comment by dirk-thomas** [comment]
While the sequence of involved libraries in the previous post is the reason it started to fail *now* the actual source of the problem is much more trivial. We have been passing a wrongly quoted `PYTHONPATH` to the test which resulted in `sys.path` to contain this weird string. And only now that value has been used to actually check if the path exists.

See ros2/rosidl_python#58 for a fix.
